src/metabase/models/field_values.clj

```diff
@@ -135,6 +221,50 @@
            (cons f (take-by-length new-length
                                    (rest s)))))))))
 
+(defn fixup-human-readable-values
+  "Field values and human readable values are lists that are zipped together. If the field values have changes, the
+  human readable values will need to change too. This function reconstructs the `human_readable_values` to reflect
+  `new-values`. If a new field value is found, a string version of that is used"
+  [{old-values :values, old-hrv :human_readable_values} new-values]
+  (when (seq old-hrv)
+    (let [orig-remappings (zipmap old-values old-hrv)]
+      (map #(get orig-remappings % (str %)) new-values))))
+
+(defn field-values->pairs
+  "Returns a list of pairs (or single element vectors if there are no human_readable_values) for the given
+  `field-values` instance."
+  [{:keys [values human_readable_values]}]
+  (if (seq human_readable_values)
+    (map vector values human_readable_values)
+    (map vector values)))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                               Advanced FieldValues                                             |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(defn advanced-fieldvalues-expired?
+  "Checks if an advanced fieldvalues expired."
+  [fv]
+  {:pre [(advanced-field-values-types (:type fv))]}
+  (u.date/older-than? (:created_at fv) (t/days 30)))
```

Comment:
Shouldn't this be referencing `advanced-field-values-max-age` above?
(And maybe make it `(def advanced-field-values-max-age (t/days 30))` instead of a raw number?)

Reply:
I changed the variable to `(t/days 30)` and referenced to it.